RestApi/Sepes.RestApi/Services/PodService.cs

```diff
@@ -36,19 +37,42 @@ public async Task Set(Pod newPod, Pod based, IEnumerable<User> newUsers, IEnumer
             var tasks = new List<Task>();
 
             // Create network security group and add rules, apply to subnet
-            bool newRulesAreAdded = based == null || 
+            bool newRulesAreAdded = based == null ||
                 !based.incoming.SequenceEqual(newPod.incoming) ||
                 !based.outgoing.SequenceEqual(newPod.outgoing);
             if (!newPod.allowAll && (newRulesAreAdded || based.allowAll)) tasks.Add(ManageNetworkSecurityGroup(newPod));
             // Removes nsg from subnet
             if (newPod.allowAll && based != null && !based.allowAll) tasks.Add(RemoveNetworkSecurityGroup(newPod));
 
-            
+
             // Add users to resources
             tasks.Add(AddUsers(newPod, newUsers, basedUsers));
 
             await Task.WhenAll(tasks);
         }
+        public async Task Delete(Pod pod)
+        {
+            //Remove nsg assignment so deletion can be done without error
+            await _azure.DeleteNetwork(pod.networkName);
+            await _azure.DeleteResourceGroup(pod.resourceGroupName);
+            var (nsgOldName, nsgNewName) = await GetNSGName(pod.networkSecurityGroupName);
+            
+            //Have to try catch so both can be attempted deleted
+            try
+            {
+                await _azure.DeleteSecurityGroup(nsgOldName);
+            }
+            catch (System.Exception)
+            {
+            }
```

Comment:
System.Exception is the base class for all exceptions so the same as catch all an anti pattern.
Done one of:
- Try to find some specific exception to use.
- Create a new Exception for azure service to throw you can test on for only that one line.
- At the very least log that it failed and then hide the error.
- If you cant handle it here but want to remove details. You can throw a new exception from catch and that will fail the task.

Reply:
Exceptions removed due to further tests showing them as unecesarry